audioipc/src/sys/unix/mod.rs

```diff
@@ -134,13 +132,13 @@ impl Drop for ConnectionBuffer {
                 "ConnectionBuffer dropped with {} bytes in cmsg",
                 self.cmsg.len()
             );
-            let b = self.cmsg.clone().freeze();
-            for fd in cmsg::iterator(b) {
-                assert_eq!(fd.len(), 1);
+            while !self.cmsg.is_empty() {
+                let fd = cmsg::decode_handle(&mut self.cmsg);
                 unsafe {
-                    close_platform_handle(fd[0]);
+                    close_platform_handle(fd);
                 }
             }
+            assert!(self.cmsg.is_empty());
```

Comment:
nit: It seems the new pieces of code in `send_msg` and `drop` can be merged into a helper

Reply:
Good idea, thanks.